File: android/app/src/main/java/com/vegettable/app/model/PriceAlert.java
package com.vegettable.app.model;

import com.google.gson.annotations.SerializedName;

public class PriceAlert {
    @SerializedName("id")
    private int id;

    @SerializedName("cropName")
    private String cropName;

    @SerializedName("targetPrice")
    private double targetPrice;

    @SerializedName("condition")
    private String condition; // "below" | "above"

    @SerializedName("isActive")
    private boolean isActive;

    @SerializedName("lastTriggeredAt")
    private String lastTriggeredAt;

    @SerializedName("createdAt")
    private String createdAt;

    public int getId() { return id; }
    public String getCropName() { return cropName; }
    public double getTargetPrice() { return targetPrice; }
    public String getCondition() { return condition; }
    public boolean isActive() { return isActive; }
    public String getLastTriggeredAt() { return lastTriggeredAt; }
    public String getCreatedAt() { return createdAt; }
}

File: android/app/src/main/java/com/vegettable/app/model/PriceAlert.kt
package com.vegettable.app.model

import com.google.gson.annotations.SerializedName

class PriceAlert {
    @SerializedName("id")
    val id: Int = 0

    @SerializedName("cropName")
    val cropName: String? = null

    @SerializedName("targetPrice")
    val targetPrice: Double = 0.0

    @SerializedName("condition")
    val condition: String? = null // "below" | "above"

    @SerializedName("isActive")
    val isActive: Boolean = false

    @SerializedName("lastTriggeredAt")
    val lastTriggeredAt: String? = null

    @SerializedName("createdAt")
    val createdAt: String? = null
}
